activation/activation.go

```diff
@@ -254,11 +256,21 @@ func (b *Builder) SmesherID() types.NodeID {
 }
 
 func (b *Builder) run(ctx context.Context) {
-	if err := b.generateInitialPost(ctx); err != nil {
+	post, metadata, err := b.generateInitialPost(ctx)
+	if err != nil {
 		b.log.Error("Failed to generate proof: %s", err)
 		return
 	}
 
+	if post != nil {
+		b.log.With().Info("initial post created", log.Object("post", post), log.Object("metadata", metadata))
+		if err := b.postSetupProvider.VerifyProof(ctx, post, metadata, verifying.WithPowCreator(b.nodeID.Bytes())); err != nil {
+			events.EmitInvalidPostProof()
+			b.log.With().Fatal("initial POST proof is invalid. Probably initialized POST data is corrupted. Please verify the data with postcli and regenerate the corrupted files.", log.Err(err))
+		}
+		b.initialPost = post
+	}
```

Comment:
Imo this should be part of `generateInitialPost` and it shouldn't return `post` and `metadata`

Reply:
>This also arguably doesn't need to be verified again

It needs to be ATM because the initial post had NOT been verified before saving to the disk (there are users with initial post created already). In the future, we can remove re-verifying it.

Reply:
Created the issue: https://github.com/spacemeshos/go-spacemesh/issues/4722